src/Aspire.Dashboard/Otlp/Storage/ApplicationKey.cs

```diff
@@ -16,7 +16,7 @@ public int CompareTo(ApplicationKey other)
         return string.Compare(InstanceId, other.InstanceId, StringComparisons.ResourceName);
     }
 
-    public bool EqualsCompositeName(string name)
+    public bool EqualsCompositeName(string? name)
```

Comment:
This is technically a breaking change, although because it is in the Dashboard we don't really care. For other src projects with Nullability issues (if any) I think it'd be best to just baseline the warnings and/or turn off nullability than trying to fix it on this PR.